tests/e2e/operator/operator_test.go

```diff
@@ -99,33 +107,58 @@ var _ = Describe("Operator", Ordered, func() {
 		})
 
 		It("deploys all the CRDs", func() {
-			Eventually(kubectl.GetCRDs).
-				Should(ContainElements(crds), "Istio CRDs are not present; expected list to contain all elements")
+			Eventually(func(g Gomega) {
+				g.Expect(k8sClient.List(context.TODO(), customResourceDefinitionList)).To(Succeed(), "Error getting CRDs list from the cluster")
+
+				var crdNames []string
+				for _, crd := range customResourceDefinitionList.Items {
+					crdNames = append(crdNames, crd.ObjectMeta.Name)
+				}
```

Comment:
Let's try to move code like this to a separate function, since it's very low level compared to the actual test. For example, this could be in a function called `getResourceNames`. And I guess we can then call this function in `WithTransform` to make the assertion a one-liner.